elements/pl-file-upload/pl-file-upload.js

```diff
@@ -1,278 +1,356 @@
 /* eslint-env browser,jquery */
-/* global _ */
-$(function () {
-  if (!window.File || !window.FileReader || !window.FileList || !window.Blob) {
-    alert(
-      'Warning: Your browser does not fully support HTML5 file upload operations.' +
-        'Please use a more current browser or you may not be able to complete this question.'
-    );
+
+(() => {
+  function escapePath(path) {
+    return path
+      .replace(/^\//, '')
+      .split('/')
+      .map((part) => encodeURIComponent(part))
+      .join('/');
   }
-});
-
-window.PLFileUpload = function (uuid, options) {
-  this.uuid = uuid;
-  this.files = options.files || [];
-  this.acceptedFiles = options.acceptedFiles || [];
-  this.acceptedFilesLowerCase = this.acceptedFiles.map((f) => f.toLowerCase());
-
-  var elementId = '#file-upload-' + uuid;
-  this.element = $(elementId);
-  if (!this.element) {
-    throw new Error('File upload element ' + elementId + ' was not found!');
+
+  /**
+   * A wrapper around fetch that return null on any error. This can simplify
+   * error handling for callers.
+   * @param {RequestInfo | URL} input
+   * @param {RequestInit} init
+   * @returns {Promise<Response | null>}
+   */
+  async function safeFetch(input, init) {
+    try {
+      const res = await fetch(input, init);
+      if (!res.ok) return null;
+      return res;
+    } catch (e) {
+      console.error(e);
+      return null;
+    }
   }
 
-  this.syncFilesToHiddenInput();
-  this.initializeTemplate();
-};
-
-/**
- * Initializes the file upload zone on the question.
- */
-window.PLFileUpload.prototype.initializeTemplate = function () {
-  var $dropTarget = this.element.find('.upload-dropzone');
-
-  var that = this;
-
-  $dropTarget.dropzone({
-    url: '/none',
-    autoProcessQueue: false,
-    accept: function (file, done) {
-      // fuzzy case:
-      const fileNameLowerCase = file.name.toLowerCase();
-      if (_.includes(that.acceptedFilesLowerCase, fileNameLowerCase)) {
-        return done();
+  class PLFileUpload {
```

Comment:
I refactored this file into a class, which unfortunately made this difficult to review. I can refactor the existing behavior in another PR to make the diff cleaner if folks would like that.

Reply:
I think we can probably just test the functionality rather than needing a careful diff? Or are there particular code changes in this file that you'd like eyes on?

Reply:
I only propose this so that I/we could reason about what actually _did_ change - the diff right now has a lot of false positives.

Reply:
Good call - that definitely helped a bunch!